### PR TITLE
#160047860 Implement Table Filters on `My Approvals` Page

### PR DESCRIPTION
**Pull request by johnmutuma5**
#### What does this PR do?
- Integrates the approvals table to the backend
  - Connects to redux state store
  - Creates actions related to fetching approvals
  - Creates an API for fetching a user's approvals with `axios`
- Implements table filters for the approvals page
- Enables pagination in the approvals table 

##### Viewing approvals
When authenticated users need to fetch requests that they are required to approve, they should be able to see them in the approvals table. 

##### Filtering approvals
They should also be able to choose to view requests that are still pending their decision to approve or decline. If they decide to view their past decisions on different requests, they should be able to do that by hitting the `Past Approvals` filter which should present them with entries that they either individually approved or declined in the past i.e. independent of the decisions made in subsequent stages of the request approval process

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into repo directory - `cd travel_tool_front`
- Check out this branch - `git checkout ft-approvals-table-filters-160047860`
- Install dependencies and run the app - `make start`
- Navigate to `localhost:3000` in a browser
- Login and navigate to the requests page
- Click the Requests dropdown item on the left-side navigation bar
- Click `My Approvals` option and get you'll be presented with requests that the logged in user is required to approve
- Click on `Pending Approvals` and you should be presented with `Open` approvals only
- Click on `Past Approvals` and you should be presented with `Approved` and `Rejected` requests only
- Click on `Items per page` and you should be able to customize the number of entries you see on each page
  - Use the `next` and `previous` buttons at the bottom of the page to navigate the pages according to you liking

#### What are the relevant pivotal tracker stories?
#160047860

#### Screenshots (if appropriate)
screen shot 2018-09-06 at 4 55 42 pm
screen shot 2018-09-06 at 4 55 50 pm
screen shot 2018-09-06 at 4 55 58 pm

#### Questions:

### CODE REVIEW

File: src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js [hunk 1]
```diff
@@ -80,36 +116,96 @@ describe('<ApprovalsPage>', () => {
...
   it('should call handleHideSearchBar method', () =>{
-    const wrapper = shallow(<Approvals />);
+    const wrapper = shallow(<Approvals {...props} />);
     wrapper.instance().handleHideSearchBar();
     expect(wrapper.state('openSearch')).toBeTruthy;
   });
 
   it('calls the onPageChange method', () => {
-    const spy = sinon.spy(Approvals.prototype, 'onPageChange');
+    // const spy = sinon.spy(Approvals.prototype, 'onPageChange');
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/__tests__/ApprovalsPage.test.js]: Please remove commented out code

File: src/views/ApprovalsPage/index.jsx [hunk 2]
```diff
@@ -1,52 +1,108 @@
...
     clickPage: true,
+    activeStatus: Utils.getActiveStatus(this.props.location.search),
+    selectedLink: 'approval page',
+    searchQuery:  this.props.location.search
   }
-  // FIX: Remove console statement and replace with actual function
-  onPageChange (page) {
-    console.log('Page Change function', page); /*eslint-disable-line */
+
+  componentWillMount () {
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/index.jsx]: Why not use `componentDidMount` here instead? From the react docs, 

_'componentDidMount() is invoked immediately after a component is mounted (inserted into the tree). Initialization that requires DOM nodes should go here. If you need to load data from a remote endpoint, this is a good place to instantiate the network request.'_

Also, `componentWillMount` is deprecated.

File: src/views/ApprovalsPage/index.jsx [hunk 3]
```diff
@@ -1,52 +1,108 @@
...
+      />
     );
   }
 
+  fetchFilteredApprovals = (query) => {
+    const { history } = this.props;
+    history.push(`/requests/my-approvals${query}`);
+  }
+
+  // FIX: Remove console statement and replace with actual function
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/index.jsx]: I believe this comment is no longer needed

File: src/views/ApprovalsPage/index.jsx [hunk 4]
```diff
@@ -1,52 +1,108 @@
...
+        />
       </div>
     );
   }
 
-  renderApprovalPage (hideSideBar,leftPaddingClass, requests, pagination, hideClass, hideClass3, selectedLink ) {
+  renderApprovalPage () {
+    const { hideNotificationPane, hideSideBar, selectedLink } = this.state;
+    let [hideClass, leftPaddingClass] = hideNotificationPane? ['hide', '']: ['', 'pd-left'];
+    const hideClass2 = hideSideBar ? 'hide mdl-cell--hide-desktop' : '';
```
Comment (amarachukwu-agbo) [inline src/views/ApprovalsPage/index.jsx]: Please can you remove lines 88 and 89 ? The variables are not being used in the renderApprovalPage method

File: src/views/ApprovalsPage/index.jsx (hunk 2 above)
Comment (johnmutuma5) [reply src/views/ApprovalsPage/index.jsx]: Great point on the deprecation @amarachukwu-agbo, plucking that one out.

File: src/views/ApprovalsPage/index.jsx (hunk 4 above)
Comment (johnmutuma5) [reply src/views/ApprovalsPage/index.jsx]: Nice. Thanks for this one too.

**Comment by johnmutuma5** [comment]
Thanks a lot @amarachukwu-agbo, for your input. Much appreciated.

**Comment by johnmutuma5** [comment]
@samuelkubai, Clean.

**Pull request merged**